### ISSUE TITLE
extend `non_constant_identifier_names` for destructured variable declarations

### ISSUE THREAD

**Issue by pq** [dart-lang/linter#3669]
Destructuring [doc](https://github.com/dart-lang/language/blob/master/working/0546-patterns/patterns-feature-specification.md#destructuring).

Example:

**BAD**

```dart
var (LAT, LONG) = geoCode('Aarhus');

(double, double) geoCode(String city) => ...;
```

**GOOD**

```dart
var (lat, long) = geoCode('Aarhus');

(double, double) geoCode(String city) => ...;
```

**Comment by pq** [comment]
This should work for [pattern-if statements](https://github.com/dart-lang/language/blob/master/working/0546-patterns/patterns-feature-specification.md#pattern-if-statement) too.

**BAD**

```dart
if (var [int X_COORD, int Y_COORD] = json) return Point(x, y);
```

**GOOD**

```dart
if (var [int xCoord, int yCoord] = json) return Point(x, y);
```

**Comment by pq** [comment]
And [pattern-for statements](https://github.com/dart-lang/language/blob/master/working/0546-patterns/patterns-feature-specification.md#pattern-for-statement):

**BAD**

```dart
var fns = <Function()>[];
for (var (FIRST, SECOND) = (0, 1); FIRST <= 13; (FIRST, SECOND) = (SECOND, FIRST + SECOND)) {
  fns.add(() {
    print(FIRST);
  });
}
```

**GOOD**

```dart
var fns = <Function()>[];
for (var (a, b) = (0, 1); a <= 13; (a, b) = (b, a + b)) {
  fns.add(() {
    print(a);
  });
}
```
